demo/Render.hpp

```diff
@@ -0,0 +1,841 @@
+     */
+    void UpdateView();
+
+    /**
+     * @brief Recalculates projection matrix
+     */
+    void UpdateProjection();
+};
+
+/**
+ * @brief Singletone class to handle input events
+ */
+class Input
+{
+public:
+    /**
+     * @brief Returns a reference to the singletone instance of the class
+     * @return reference to the instance
+     */
+    static Input& GetInstance();
+
+    /**
+     * @brief Initializes input with the given contex
+     * @param[in] window pointer to the glfw window for which callbacks are set
+     */
+    static void InitializeContext(GLFWwindow* window);
+
+    /**
+     * @brief Push additional callback for the Resize event
+     *
+     * @attention the callback must be c-style function
+     * @param callback function object containing the pointer to the callback function
+     */
+    void AddResizeCallback(std::function<void(GLFWwindow*, int, int)> callback);
+
+    /**
+     * @brief Removes given callback function from the Resize event call list
+     *
+     * @attention the callback must be c-style
+     * @param callback pointer to the callback function to be removed
+     */
+    void RemoveResizeCallback(std::function<void(GLFWwindow*, int, int)> callback);
+
+    /**
+     * @brief Push additional callback for the KeyButton event
+     *
+     * @attention the callback must be c-style function
+     * @param callback function object containing the pointer to the callback function
+     */
+    void AddKeyButtonCallback(std::function<void(GLFWwindow*, int, int, int, int)> callback);
+
+    /**
+     * @brief Removes given callback function from the KeyButton event call list
+     *
+     * @attention the callback must be c-style
+     * @param callback pointer to the callback function to be removed
+     */
+    void RemoveKeyButtonCallback(std::function<void(GLFWwindow*, int, int, int, int)> callback);
+
+    /**
+     * @brief Push additional callback for the CursoreMove event
+     *
+     * @attention the callback must be c-style function
+     * @param callback function object containing the pointer to the callback function
+     */
+    void AddCursoreMoveCallback(std::function<void(GLFWwindow*, double, double)> callback);
+
+    /**
+     * @brief Removes given callback function from the CursoreMove event call list
+     *
+     * @attention the callback must be c-style
+     * @param callback pointer to the callback function to be removed
+     */
+    void RemoveCursoreMoveCallback(std::function<void(GLFWwindow*, double, double)> callback);
+
+    /**
+     * @brief Push additional callback for the MouseButton event
+     *
+     * @attention the callback must be c-style function
+     * @param callback function object containing the pointer to the callback function
+     */
+    void AddMouseButtonCallback(std::function<void(GLFWwindow*, int, int, int)> callback);
+
+    /**
+     * @brief Removes given callback function from the MouseButton event call list
```

Comment:
you keep mixing tenses in documentation. In one you `Push` but in the other you `Removes`